targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF53/i2c_api.c

```diff
@@ -196,4 +471,123 @@ int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop)
     return obj->transfer_result;
 }
 
+#if DEVICE_I2CSLAVE
+
+/** Configure I2C as slave or master.
+ *  @param obj_ The I2C object
+ *  @param enable_slave Enable i2c hardware in slave mode
+ *  @return non-zero if a value is available
+ */
+void i2c_slave_mode(i2c_t *obj_, int enable_slave)
+{
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
+    /* Reconfigure this instance as an I2C slave */
+    obj->is_slave = enable_slave;
+
+    if (obj->is_slave && !obj->slave_instance.p_reg) {
+        obj->slave_instance = get_next_free_twis_instance();
+    }
+
+    i2c_configure_driver_instance(obj_);
+}
+
+/** Check to see if the I2C slave has been addressed.
+ *  @param obj The I2C object
+ *  @return The status - 1 - read addresses, 2 - write to all slaves,
+ *         3 write addressed, 0 - the slave has not been addressed
+ */
+int i2c_slave_receive(i2c_t *obj_)
+{
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
+    /* return if slave instance is invalid */
+    if (!obj->slave_instance.p_reg)
+        return 0;
+
+    if (nrfx_twis_is_waiting_rx_buff(&obj->slave_instance)) {
+        return 3;
+    }
+
+    if (nrfx_twis_is_waiting_tx_buff(&obj->slave_instance)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+/** Configure I2C as slave or master.
+ *  @param obj The I2C object
+ *  @param data    The buffer for receiving
+ *  @param length  Number of bytes to read
+ *  @return non-zero if a value is available
+ */
+int i2c_slave_read(i2c_t *obj_, char *data, int length)
+{
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
+    /* return if slave instance is invalid */
+    if (!obj->slave_instance.p_reg)
+        return 0;
+
+    const nrfx_twis_t *twis_instance = &obj->slave_instance;
+
+    /* Wait until the master is trying to write data */
+    while (!nrfx_twis_is_waiting_rx_buff(twis_instance)) { }
```

Comment:
timeout?

Reply:
Follows the implementation of the master mode so I think it is OK